### DISCUSSION TITLE
/gridpoints 301: Moved Permanently

### DISCUSSION THREAD

**Discussion by carolynzech** [weather-gov/api#599] [Q&A]
I'm trying to query the /gridpoints endpoint on my Arduino, but I'm getting an error that the location has moved permanently. Attaching the relevant code and screenshot of the error below--any help would be much appreciated.

```
bool connect_to_webpage() {
  if (client.connect("api.weather.gov", 80)) {
    client.println("GET /gridpoints/BOX/64,64/forecast HTTP/1.1");
    client.println("User-Agent: cs1600 project");
    client.println("Host: api.weather.gov");
    client.println("accept: application/geo+json");
    client.println("Connection: keep-alive");
    client.println();
    return true;
  } else {
    Serial.println("Failed to fetch webpage");
    return false;
  }
}
```
Getting this error:
<img width="541" alt="Screenshot 2022-12-01 at 3 21 52 PM" src="https://user-images.githubusercontent.com/71352687/205151838-3b9c7bb7-43fe-4b11-9511-6399b4f55b6d.png">

**Comment by netbymatt** [comment]
I'm not sure what language you're writing in but by guess is that this line:
```
 if (client.connect("api.weather.gov", 80)) {
```
is causing it to connect to port 80 (HTTP) and the 304 redirect is pointing you to port 443 (HTTPS).

Just changing to 443 probably won't be enough. Whatever is declaring client outside of the code you provided will probably have to use an HTTPS equivalent as well.
